draft-ietf-masque-h3-datagram.md

```diff
@@ -194,36 +203,49 @@ recent draft version from the intersection set. This ensures that both
+16.7 of !HTTP=I-D.ietf-httpbis-semantics}}) can choose to use; this includes the
+HTTP Upgrade mechanism (see {{Section 16.7 of
+!HTTP=I-D.ietf-httpbis-semantics}}) and extensions to the CONNECT method (for
+example, see {{?EXT-CONNECT=RFC8441}}). It allows endpoints to reliably
+communicate request-related information end-to-end on HTTP request streams, even
+in the presence of HTTP intermediaries. The Capsule Protocol can be used to
+exchange HTTP Datagrams, which is necessary when HTTP is running over a
 transport that does not support the QUIC DATAGRAM frame.
 
+
+## HTTP Data Streams {#data-stream}
+
 This specification defines the "data stream" of an HTTP request as the
-bidirectional stream of bytes that follow the headers in both directions. In
-HTTP/1.x, the data stream consists of all bytes on the connection that follow
-the blank line that concludes either the request header section, or the 2xx
-(Successful) response header section. (Note that only a single HTTP request
-starting the capsule protocol can be sent on HTTP/1.x connections.) In HTTP/2
-and HTTP/3, the data stream of a given HTTP request consists of all bytes sent
-in DATA frames with the corresponding stream ID. The concept of a data stream is
-particularly relevant for methods such as CONNECT where there is no HTTP message
-content after the headers.
+bidirectional stream of bytes that follows the header section of the request
+message and the final, successful (i.e., 2xx) response message.
 
-Note that use of the Capsule Protocol is not required to use HTTP Datagrams. If
-a new HTTP Upgrade Token is only defined over transports that support QUIC
-DATAGRAM frames, they might not need a stream encoding. Additionally,
-definitions of new HTTP Upgrade Tokens can use HTTP Datagrams with their own
-data stream protocol. However, new HTTP Upgrade Tokens that wish to use HTTP
-Datagrams SHOULD use the Capsule Protocol unless they have a good reason not to.
+In HTTP/1.x, the data stream consists of all bytes on the connection that follow
+the blank line that concludes either the request header section, or the response
+header section. As a result, only a single HTTP request starting the capsule
+protocol can be sent on HTTP/1.x connections.
+
+In HTTP/2 and HTTP/3, the data stream of a given HTTP request consists of all
+bytes sent in DATA frames with the corresponding stream ID.
+
+The concept of a data stream is particularly relevant for methods such as
+CONNECT where there is no HTTP message content after the headers.
+
+Data streams can be prioritized using any means suited to stream or request
+prioritization. For example, see {{Section 11 of
+?PRIORITY=I-D.ietf-httpbis-priority}}.
 
 
-## Capsule Protocol {#capsule-protocol}
+## The Capsule Protocol {#capsule-protocol}
 
 Definitions of new HTTP Upgrade Tokens can state that their data stream uses the
 Capsule Protocol. If they do so, that means that the contents of their data
```

Comment:
This is very abrupt; it assumes the reader knows that upgrade tokens can be encountered in the upgrade mechanism or extended CONNECT. The reader also has to assume the relationships therein -- e.g., if I define foo to use the Capsule Protocol, does it mean that it it'll use it whether the upgrade token is encountered in extended CONNECT or in Upgrade?

It might be better to NOT focus on upgrade tokens so much as the primary mechanism here and below; after all, they're just identifiers.

Reply:
Fixed by introducing what HTTP Upgrade tokens are when we first use the term